packages/lambda-xyz/src/tile.set.cache.ts

```diff
@@ -9,20 +9,88 @@ export function getTileSet(name: string, projection: Epsg): TileSet | undefined
     return TileSets.get(tileSetId);
 }
 
-export async function loadTileSet(req: LambdaContext, name: string, projection: Epsg): Promise<TileSet | null> {
-    req.set('tileSet', name);
-    req.set('projection', projection);
+export async function loadTileSet(name: string, projection: Epsg): Promise<TileSet | null> {
     let tileSet = getTileSet(name, projection);
     if (tileSet == null) {
         tileSet = new TileSet(name, projection);
         TileSets.set(tileSet.id, tileSet);
     }
-    req.timer.start('tileset:load');
     const loaded = await tileSet.load();
-    req.timer.end('tileset:load');
     if (!loaded) {
         TileSets.delete(tileSet.id);
         return null;
     }
     return tileSet;
 }
+
+class IndividualTileSet extends TileSet {
+    constructor(parent: TileSet, image: TileSetRuleImagery, projection: Epsg) {
+        super(image.imagery.id, projection);
+        // use parent data as prototype for child;
+        this.tileSet = Object.create(parent.tileSet ?? null);
+
+        // Override title
+        this.tileSet.title = `${parent.name}_${image.imagery.name}`;
+        const rule = {
+            id: image.imagery.id,
+            minZoom: 0,
+            maxZoom: image.rule.maxZoom,
+            priority: 0,
+        };
+        // Override imagery (for completeness)
+        this.tileSet.imagery = { [rule.id]: rule };
+
+        this.imagery = [
+            {
+                rule,
+                imagery: image.imagery,
+            },
+        ];
+    }
+}
+
+async function collectTileSets(tileSets: TileSet[], name: string, projection: Epsg | null): Promise<void> {
+    if (projection == null) {
+        for (const code of Epsg.TargetCodes) {
+            collectTileSets(tileSets, name, Epsg.get(code));
+        }
+        return;
+    }
+
+    const tileSet = await loadTileSet(name, projection);
+    if (tileSet != null) {
+        tileSets.push(tileSet);
+
+        if (tileSet.imagery != null) {
+            for (const image of tileSet.imagery) {
+                tileSets.push(new IndividualTileSet(tileSet, image, projection));
+            }
+        }
+    }
+}
+/**
+ * Load a collection of TileSets. Used by WMTSCapabilities
+
+ * @param name if an empty string load all TileSets
+ * @param projection if null load all projections
+ */
+export async function loadTileSets(name: string, projection: Epsg | null): Promise<TileSet[]> {
+    if (name !== '' && projection != null) {
+        // single tileSet
+        const ts = await loadTileSet(name, projection);
+
+        return ts == null ? [] : [ts];
+    }
+
+    const tileSets: TileSet[] = [];
+
+    if (name === '') {
+        for (const tsn in TileSetName) {
```

Comment:
`for in` is dangerous with enums, as sometimes they will be doubly linked eg here is the generated source for EpsgCode enum

```
    {
  '2193': 'Nztm2000',
  '3793': 'Citm2000',
  '3857': 'Google',
  '4326': 'Wgs84',
  Google: 3857,
  Wgs84: 4326,
  Nztm2000: 2193,
  Citm2000: 3793
}
```